flake.nix

```
@@ -339,7 +339,7 @@
           '';
         }; in
 	pkgs.mkShell {
-          buildInputs = [ updateCmd ];
+          buildInputs = [ updateCmd pkgs.haskell.compiler.ghc810 pkgs.cabal-install pkgs.zlib pkgs.gmp ];
```

```suggestion
          buildInputs = [ updateCmd pkgs.haskell.compiler.ghc810 pkgs.cabal-install pkgs.zlib pkgs.gmp pkgs.openssl ];
```

I believe I also needed openssl as I didn't have that globally. I'm not sure, but `pkg-config` might also be needed to find the openssl headers.